android/src/main/java/com/swmansion/gesturehandler/core/LongPressGestureHandler.kt

```diff
@@ -53,43 +91,72 @@ class LongPressGestureHandler(context: Context) : GestureHandler<LongPressGestur
       previousTime = SystemClock.uptimeMillis()
       startTime = previousTime
       begin()
-      startX = sourceEvent.rawX
-      startY = sourceEvent.rawY
+      startX = getAverageX(sourceEvent)
+      startY = getAverageY(sourceEvent)
+
+      ++currentPointers
     }
 
-    if (state == STATE_BEGAN && sourceEvent.pointerCount == numberOfPointersRequired) {
+    if (sourceEvent.actionMasked == MotionEvent.ACTION_POINTER_DOWN) {
+      startX = getAverageX(sourceEvent)
+      startY = getAverageY(sourceEvent)
+
+      if (++currentPointers > numberOfPointersRequired) {
+        fail()
+        currentPointers = 0
+      }
+    }
+    if (state == STATE_BEGAN && currentPointers == numberOfPointersRequired && !hasScheduledActivation) {
       handler = Handler(Looper.getMainLooper())
       if (minDurationMs > 0) {
         handler!!.postDelayed({ activate() }, minDurationMs)
       } else if (minDurationMs == 0L) {
         activate()
       }
+
+      hasScheduledActivation = true
```

Comment:
Is this property necessary? Wouldn't checking whether action is `DOWN` or `POINTER_DOWN` accomplish the same?

Reply:
Yes, it should work the same.

Replaced in [04b4c9f](https://github.com/software-mansion/react-native-gesture-handler/pull/3043/commits/04b4c9fba8269465b499daf8ecfea70f59a2ab4f)